server/aws-lsp-codewhisperer/src/language-server/utilities/supplementalContextUtil/supplementalContextUtil.ts

```diff
@@ -0,0 +1,73 @@
+// Port of implementation in AWS Toolkit for VSCode
+// https://github.com/aws/aws-toolkit-vscode/blob/9d8ddbd85f4533e539a58e76f7c46883d8e50a79/packages/core/src/codewhisperer/util/supplementalContext/supplementalContextUtil.ts
+
+import { fetchSupplementalContextForSrc } from './crossFileContextUtil'
+import { isTestFile } from './codeParsingUtil'
+import { CodeWhispererSupplementalContext } from '../../models/model'
+import {
+    CancellationToken,
+    Logging,
+    Position,
+    TextDocument,
+    Workspace,
+} from '@aws/language-server-runtimes/server-interface'
+
+export class CancellationError extends Error {}
+
+export function fetchSupplementalContext(
+    document: TextDocument,
+    position: Position,
+    workspace: Workspace,
+    logging: Logging,
+    cancellationToken: CancellationToken
+): CodeWhispererSupplementalContext | undefined {
+    const timesBeforeFetching = performance.now()
+
+    const isUtg = isTestFile(document.uri, {
```

Comment:
I don't have full context, what does UTG actually stand for?

Reply:
I think it stands for "unit test generator". I don't know how significant this meaning is, tbh.